diff --git a/include/my_global.h b/include/my_global.h
index ac5d722..ee63d16 100644
--- a/include/my_global.h
+++ b/include/my_global.h
@@ -132,10 +132,10 @@
 #  define SIZEOF_INT 4
 #  define SIZEOF_LONG_LONG 8
 #  define SIZEOF_OFF_T 8
-#  if defined(__i386__) || defined(__ppc__)
+#  if defined(__i386__) || defined(__ppc__) || defined(__arm__)
 #    define SIZEOF_CHARP 4
 #    define SIZEOF_LONG 4
-#  elif defined(__x86_64__) || defined(__ppc64__)
+#  elif defined(__x86_64__) || defined(__ppc64__) || defined(__arm64__)
 #    define SIZEOF_CHARP 8
 #    define SIZEOF_LONG 8
 #  else
diff --git a/include/my_net.h b/include/my_net.h
index 3af79ea..26bc28e 100644
--- a/include/my_net.h
+++ b/include/my_net.h
@@ -44,9 +44,9 @@ C_MODE_START
 #endif
 
 #if !defined(__WIN__) && !defined(HAVE_BROKEN_NETINET_INCLUDES) && !defined(__BEOS__) && !defined(__NETWARE__)
-#include <netinet/in_systm.h>
+/* #include <netinet/in_systm.h> */
 #include <netinet/in.h>
-#include <netinet/ip.h>
+/* #include <netinet/ip.h> */
 #if !defined(alpha_linux_port)
 #include <netinet/tcp.h>
 #endif
